diff --git a/lib/worker.cjs b/lib/worker.cjs
index 14f4584c748bbec5a1a0282afe486c098668dfe6..172eedec148df0a78057bf4ae095387ddde2f8bb 100644
--- a/lib/worker.cjs
+++ b/lib/worker.cjs
@@ -22,4 +22 @@
-	const bytes = Buffer.from(koffi.view(address, 32768));
-	let end = 0;
-	while (end + 1 < bytes.length && bytes[end] !== 0) end += 2;
-	return bytes.toString("utf16le", 0, end);
+	return koffi.decode.string16(address);
@@ -208 +205 @@
-const post = (message) => {
+const post = (message, terminal = false) => {
@@ -211 +208 @@
-		if (process.connected) process.disconnect();
+		if (terminal && process.connected) process.disconnect();
@@ -232 +229 @@
-		});
+		}, true);
@@ -237 +234 @@
-		});
+		}, true);
